scripts/config

```diff
@@ -11,153 +11,10 @@ source /usr/share/yunohost/helpers
-    port=$(ynh_app_setting_get --app $app --key synapse_port)
-    synapse_tls_port=$(ynh_app_setting_get --app $app --key synapse_tls_port)
-    turnserver_tls_port=$(ynh_app_setting_get --app $app --key turnserver_tls_port)
-    turnserver_pwd=$(ynh_app_setting_get --app $app --key turnserver_pwd)
-    registration_shared_secret=$(ynh_app_setting_get --app $app --key registration_shared_secret)
-    form_secret=$(ynh_app_setting_get --app $app --key form_secret)
-    macaroon_secret_key=$(ynh_app_setting_get --app=$app --key=macaroon_secret_key)
-
-    synapse_user="matrix-$app"
-    synapse_db_name="matrix_$app"
-    synapse_db_user="matrix_$app"
-
-    # Configure Synapse
-
-    # WARNING : theses command are used in INSTALL, UPGRADE, CONFIG, CHANGE-URL (4 times)
-    # For any update do it in all files
-
-    if [ -z $macaroon_secret_key ]; then
-    # Well, in this package this value was not managed because it was not needed, synapse is able to generate this with some other secret in the config file but after some vulnerability was found with this practice.
-    # For more detail about this issue you can see : https://matrix.org/blog/2019/01/15/further-details-on-critical-security-update-in-synapse-affecting-all-versions-prior-to-0-34-1-cve-2019-5885/
-    # The problem is that we can't just say generate a new value if the package has not already defined a value. The reason is that changing this value logout all user. And in case of a user has enabled the encryption, the user might lost all conversation !!
-    # So for the old install we just leave this as it is. And for the new install we use a real macaroon.
-    macaroon_secret_key_param='# macaroon_secret_key: ""'
-    else
-        macaroon_secret_key_param='macaroon_secret_key: "'$macaroon_secret_key'"'
-    fi
-
-    if [ $is_free_registration -eq 0 ]
-    then
-        allowed_access=False
-        sso_enabled=True
-    else
-        allowed_access=True
-        sso_enabled=False
-    fi
-
-    ynh_add_config --template="homeserver.yaml" --destination="/etc/matrix-$app/homeserver.yaml"
-    ynh_add_config --template="log.yaml" --destination="/etc/matrix-$app/log.yaml"
-
-    # Create .well-known redirection for access by federation
-    if yunohost --output-as plain domain list | grep -q "^$server_name$"
-    then
-        ynh_add_config --template="server_name.conf" --destination="/etc/nginx/conf.d/${server_name}.d/${app}_server_name.conf"
-    fi
-
-    ynh_store_file_checksum --file "$homeserver_config_path"
-    setfacl -R -m user:turnserver:rX  /etc/matrix-$app
-
-    systemctl restart matrix-$app
-    systemctl reload nginx
-}
 
 #=================================================
 # GENERIC FINALIZATION
 #=================================================
 # SELECT THE ACTION FOLLOWING THE GIVEN ARGUMENT
 #=================================================
-case $1 in
-    show) show_config;;
-    apply) apply_config;;
-esac
+ynh_app_config_run $1
```

Comment:
Are you sure that after running this you don't need to restart marix-synapse and nginx  ?

Reply:
this is done automatically with `services = ["matrix-__APP__"]` defined in `config_panel.toml`. I asked how to restart 2 services